server/src/main/java/io/spine/server/event/enrich/Linker.java

```diff
@@ -0,0 +1,247 @@
+            Set<FieldDescriptor> sourceFields = toSourceFields(enrichmentField);
+            putEnrichmentsByField(enrichmentField, sourceFields);
+        }
+
+        ImmutableList<EnrichmentFunction<?, ?, ?>> functions = this.functions.build();
+        ImmutableMultimap<FieldDescriptor, FieldDescriptor> fields = this.fields.build();
+        //TODO:2019-02-02:alexander.yevsyukov: Enable the below checks when enrichment schemas are generated per bounded context.
+        //checkFunctions(functions);
+        //checkFields(fields);
+
+        FieldTransitions result = new FieldTransitions(functions, fields);
+        return result;
+    }
+
+    @SuppressWarnings("unused")
+    private void checkFunctions(ImmutableList<EnrichmentFunction<?, ?, ?>> functions) {
+        checkState(
+                !functions.isEmpty(),
+                "No functions found for creating enrichment of type `%s`" +
+                " by the source message of the type `%s`." +
+                " Please check `EnricherBuilder.add()` is called with corresponding" +
+                " source/target field class arguments.",
+                enrichmentDescriptor.getFullName(),
+                sourceDescriptor.getFullName()
+        );
+    }
+
+    @SuppressWarnings("unused")
+    private void checkFields(ImmutableMultimap<FieldDescriptor, FieldDescriptor> fields) {
+        checkState(!fields.isEmpty(),
+                   "Unable to match fields for enriching `%s` with `%s`",
+                   sourceDescriptor.getFullName(), enrichmentDescriptor.getFullName());
+    }
+
+    private void putEnrichmentsByField(FieldDescriptor enrichmentField,
+                                       Iterable<FieldDescriptor> sourceFields) {
+        for (FieldDescriptor sourceField : sourceFields) {
+            Optional<FieldEnrichment<?, ?, ?>> found = transition(sourceField, enrichmentField);
+            found.ifPresent(fn -> {
+                functions.add(fn);
+                fields.put(sourceField, enrichmentField);
+            });
+        }
+    }
+
+    private Set<FieldDescriptor> toSourceFields(FieldDescriptor enrichmentField) {
+        ImmutableList<FieldRef> fieldReferences = allFrom(enrichmentField.toProto());
+        checkState(!fieldReferences.isEmpty(),
+                   "Unable to get source field information from the enrichment field `%s`",
+                   enrichmentField.getFullName());
+
+        return findSourceFields(fieldReferences, enrichmentField);
+    }
+
+    private Set<FieldDescriptor>
+    findSourceFields(ImmutableList<FieldRef> references, FieldDescriptor enrichmentField) {
+        int refCount = references.size();
+        checkArgument(refCount > 0, "References may not be empty");
+        Set<FieldDescriptor> result = new HashSet<>(refCount);
+
+        FieldDescriptor.Type basicType = null;
+        Descriptor messageType = null;
+        for (FieldRef ref : references) {
+            boolean strict = refCount == 1;
```

Comment:
I guess `strict` should be taken out of the `for` cycle. It is not modified anywhere and the `refCount` is also a constant.